pkg/controllers/management/clusterprovisioner/rke_store.go

```diff
@@ -1,48 +1,63 @@
 package clusterprovisioner
 
 import (
-	"github.com/rancher/kontainer-driver-metadata/rke/templates"
+	"strings"
+
 	kontainerengine "github.com/rancher/kontainer-engine/drivers/rke"
 	kd "github.com/rancher/rancher/pkg/controllers/management/kontainerdrivermetadata"
+	"github.com/rancher/rancher/pkg/namespace"
 	v3 "github.com/rancher/types/apis/management.cattle.io/v3"
 	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-var addonsList = []string{
-	templates.Calico,
-	templates.Flannel,
-	templates.Weave,
-	templates.Canal,
-	templates.CoreDNS,
-	templates.KubeDNS,
-	templates.MetricsServer,
-	templates.NginxIngress}
-
 type rkeStore struct {
-	AddonLister     v3.RKEAddonLister
-	Addons          v3.RKEAddonInterface
-	SvcOptionLister v3.RKEK8sServiceOptionLister
-	SvcOptions      v3.RKEK8sServiceOptionInterface
+	AddonLister        v3.RKEAddonLister
+	Addons             v3.RKEAddonInterface
+	SvcOptionLister    v3.RKEK8sServiceOptionLister
+	SvcOptions         v3.RKEK8sServiceOptionInterface
+	SystemImagesLister v3.RKEK8sSystemImageLister
+	SystemImages       v3.RKEK8sSystemImageInterface
 }
 
-func NewDataStore(addonLister v3.RKEAddonLister, addons v3.RKEAddonInterface, svcOptionLister v3.RKEK8sServiceOptionLister, svcOptions v3.RKEK8sServiceOptionInterface) kontainerengine.Store {
+func NewDataStore(addonLister v3.RKEAddonLister, addons v3.RKEAddonInterface,
+	svcOptionLister v3.RKEK8sServiceOptionLister, svcOptions v3.RKEK8sServiceOptionInterface,
+	sysImageLister v3.RKEK8sSystemImageLister, sysImages v3.RKEK8sSystemImageInterface) kontainerengine.Store {
 	return &rkeStore{
-		AddonLister:     addonLister,
-		Addons:          addons,
-		SvcOptionLister: svcOptionLister,
-		SvcOptions:      svcOptions,
+		AddonLister:        addonLister,
+		Addons:             addons,
+		SvcOptionLister:    svcOptionLister,
+		SvcOptions:         svcOptions,
+		SystemImagesLister: sysImageLister,
+		SystemImages:       sysImages,
 	}
 }
 
 func (a *rkeStore) GetAddonTemplates(k8sVersion string) map[string]interface{} {
 	data := map[string]interface{}{}
-	for _, addonName := range addonsList {
-		template, err := kd.GetRKEAddonTemplate(k8sVersion, addonName, a.AddonLister, a.Addons)
+	sysImage, err := a.SystemImagesLister.Get(namespace.GlobalNamespace, k8sVersion)
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			logrus.Errorf("getAddonTemplates: error finding system image for %s %v", k8sVersion, err)
+			return data
```

Comment:
Don't we want to return the error?

Reply:
this is getting called from kontainer engine to get data, in this case, we just log the error and move on with the assumption that either rke will have the template or rke will fail with proper msg